Refactor syntax unicode findname

livecodefraser opened the pull request:
Depends on refactor-syntax_unicode_findstasck

Commit:
Plumb in the kMCPropertyTypeName property type

The property getter/setter part of the engine had no idea how to deal with properties of this type, causing strange failures in the IDE (e.g. the right-click menu for buttons not appearing).

Commit:
Merge branch 'refactor-syntax_unicode_findstack' into refactor-syntax_unicode_findname

Conflicts:
	engine/src/dispatch.cpp

Commit:
Change the various {find,get*}name methods to use NameRefs

In the process, MCObject::names now returns a ValueRef instead of a StringRef.

Commit:
Fix a couple of oversights creating link errors

A couple of functions had incorrect signatures.

livecodesebastien commented on engine/src/stack.cpp:
```diff
@@ -2795,28 +2795,33 @@ Boolean MCStack::del()
-			delete t_new_name;
 		}
-		setname_cstring(t_new_name);
+		setname(t_name);
```
t_name should be deleted after that as setname() increases the number of references.

livecodesebastien commented on engine/src/stack.cpp:
```diff
@@ -2795,28 +2795,33 @@ Boolean MCStack::del()
 			else
-				sprintf(t_new_name, "Copy (%d) of %s", t_index, t_old_name);
-			if (MCdispatcher -> findstackname(t_new_name) == NULL)
+				/* UNCHECKED */ MCStringFormat(&t_new_name, "Copy (%d) of %@", t_index, *t_new_name);
```
The operation "&t_new_name" will fail on a second iteration, as the member m_value of t_new_name won't be NULL after the first iteration - see operator& of MCAutoValueRef (MCAssert would fail)

Commit:
Fix a couple of bugs in MCStack::paste

Pasting stacks suffered from a memory leak. Additionally, it would have failed an assert if the pasted stack had to be renamed to "Copy (<n>) of ..." in order to avoid name clashes.

Commit:
Merge branch 'refactor-syntax_unicode' into refactor-syntax_unicode_findname

Conflicts:
	engine/src/stack2.cpp

Commit:
Fix a number of errors

Commit:
Merge branch 'refactor-syntax_unicode_findstack' into refactor-syntax_unicode_findname

Conflicts:
	engine/src/dispatch.cpp

livecodesebastien commented:
Can you merge `refactor-syntax_unicode_findstack` in (after having merged `refactor-syntax_unicode` in it)?

Commit:
Merge branch 'refactor-syntax_unicode_findstack' into refactor-syntax_unicode_findname

livecodesebastien commented:
Can you merge refactor-syntax_unicode in?

Commit:
Merge branch 'refactor-syntax_unicode' into refactor-syntax_unicode_findname

Conflicts:
	engine/src/exec.h
	engine/src/objectprops.cpp

Commit:
Merge remote-tracking branch 'upstream/refactor-syntax_unicode' into refactor-syntax_unicode_findname

Conflicts:
	engine/src/exec-interface.cpp

Commit:
Merge remote-tracking branch 'upstream/refactor-syntax_unicode' into refactor-syntax_unicode_findname

Conflicts:
	engine/src/objectprops.cpp

Commit:
Merge remote-tracking branch 'upstream/refactor-syntax_unicode' into refactor-syntax_unicode_findname

livecodesebastien commented:
There are merging conflicts for now due to the last merging in _unicode, can you merge it in?

Commit:
Merge remote-tracking branch 'upstream/refactor-syntax_unicode' into refactor-syntax_unicode_findname

Conflicts:
	engine/src/dispatch.cpp
	engine/src/exec-debugging.cpp
